data_adapter_oemof/build_datapackage.py

```diff
@@ -384,13 +407,75 @@ def yearly_scalars_to_periodic_values(scalar_dataframe) -> None:
         )
         return scalar_dataframe
 
+    def time_series_aggregation(self, tsam_config: str, destination: str = None):
+        """
+        Aggregates time series in datapackage and saves the new datapackage with updated
+        (sequence)Resources as well as aggregated sequence resources.
+        Parameters
+        ----------
+        tsam_config
+        destination
+
+        Returns
+        -------
+
+        """
+        # Refactor sequences into one Dataframe
+        self.save_datapackage_to_csv(location_to_save_to=destination)
+        sequences = pd.concat(
+            self.parametrized_sequences.values(),
+            axis=1,
+            keys=self.parametrized_sequences.keys(),
+        )
+        sequences.index = self.periods.index
+        sequences["periods"] = self.periods.periods
+
+        # Group sequences by Periods
+        tsam_aggregated_typical_periods = []
+        for period, period_sequence in sequences.groupby(by="periods"):
+            # Saving the old Index to have it for later periods creation
+            index_old = period_sequence.index
+            # Throw away data that might disturb aggregation
+            period_sequence.drop(["periods"], axis=1, inplace=True)
+            # Aggregate
+            aggregation = tsam.TimeSeriesAggregation(period_sequence, **tsam_config)
```

Comment:
Attention: The idea is to allow different TSAM configs for each period, thus tsam_config should be a list of configs for each period (or you could allow having a single TSAM config which is used for all periods as well)